shared/src/main/scala/org/adridadou/openlaw/oracles/ExternalCallOracle.scala

```diff
@@ -94,13 +143,16 @@ object SuccessfulExternalCallEvent {
 }
 
 sealed trait ExternalCallEvent extends OpenlawVmEvent {
-  val identifier:ActionIdentifier
+  val contractId: ContractId
```

Comment:
for the external call event, you don't need to add them here. You can get the contract id from the VM

vm.contractDefinition.id(crypto) will give you the contract id linked to this VM

This makes me think, we will need to make it possible to send an id that is not a contract id in the future if we want to use this in a flow as well. Something to think about. 

So actually, maybe let's not remove the contractId from here but let's change the type to something like CallerId or something like that.

For a contract, callerId = contractId, for the flow, it will be the flowExecutionId etc ... 

WDYT ?

Reply:
That makes sense to me. I will update that accordingly. Thanks